app/graphql/types/rule.rb

```diff
@@ -23,41 +23,65 @@ class Rule < Types::BaseObject
                required: false
     end
 
+    # rubocop:disable Metrics/AbcSize
     def compliant(args = {})
-      system_id(args) &&
-        profile_id(args) &&
-        %w[pass notapplicable notselected].include?(
-          context[:rule_results][object.id][profile_id(args)]
-        )
+      ::Rails.cache.fetch(host: system_id(args), profile: profile_id(args),
+                          rule: object.id, attribute: 'compliant') do
+        system_id(args) && profile_id(args) &&
+          %w[pass notapplicable notselected].include?(
+            context[:rule_results][object.id][profile_id(args)]
+          )
+      end
+    end
+    # rubocop:enable Metrics/AbcSize
+
+    def cached_references
+      ::Rails.cache.read(rule: object.id, attribute: 'references')
     end
 
+    # rubocop:disable Metrics/AbcSize
     def references
+      return cached_references if cached_references
+
       if context[:"rule_references_#{object.id}"].nil?
         ::CollectionLoader.for(::Rule, :rule_references)
                           .load(object).then do |references|
-          references.map { |ref| [ref.href, ref.label] }.to_json
+          references.map { |ref| { href: ref.href, label: ref.label } }.to_json
```

Comment:
Notice I'm bringing this back in because current frontend-components isn't working: if we get it in, we can save this https://github.com/RedHatInsights/frontend-components/pull/471